src/H5encode.h

```diff
@@ -0,0 +1,283 @@
+    }
+
+#define UINT64ENCODE(p, n)                                                                                   \
+    {                                                                                                        \
+        uint64_t _n = (n);                                                                                   \
+        size_t   _i;                                                                                         \
+        uint8_t *_p = (uint8_t *)(p);                                                                        \
+                                                                                                             \
+        for (_i = 0; _i < sizeof(uint64_t); _i++, _n >>= 8)                                                  \
+            *_p++ = (uint8_t)(_n & 0xff);                                                                    \
+        for (/*void*/; _i < 8; _i++)                                                                         \
+            *_p++ = 0;                                                                                       \
+        (p) = (uint8_t *)(p) + 8;                                                                            \
+    }
+
+/* Encode a 64-bit unsigned integer into a variable-sized buffer */
+/* (Assumes that the high bits of the integer are zero) */
+#define UINT64ENCODE_VAR(p, n, l) ENCODE_VAR(p, uint64_t, n, l)
+
+#define H5_ENCODE_UNSIGNED(p, n)                                                                             \
+    {                                                                                                        \
+        HDcompile_assert(sizeof(unsigned) == sizeof(uint32_t));                                              \
+        UINT32ENCODE(p, n)                                                                                   \
+    }
+
+/* Assumes the endianness of uint64_t is the same as double */
+#define H5_ENCODE_DOUBLE(p, n)                                                                               \
+    {                                                                                                        \
+        uint64_t _n;                                                                                         \
+        size_t   _u;                                                                                         \
+        uint8_t *_p = (uint8_t *)(p);                                                                        \
+                                                                                                             \
+        HDcompile_assert(sizeof(double) == 8);                                                               \
+        HDcompile_assert(sizeof(double) == sizeof(uint64_t));                                                \
+        HDmemcpy(&_n, &n, sizeof(double));                                                                   \
```

Comment:
Could cause a strict-alias issue, but not worth holding this PR as I think this is rarely used.